### PR TITLE
basic auth functionality

### PR DESCRIPTION
**Pull request by nonsense**
Partially addresses https://github.com/testground/testground/issues/1112

We will need something like this, once we spin up ingress for the daemon in the CI cluster.

---

FOLLOW UP TODO:
- [ ] maybe add another X-header that is `username` which will be attached to `jobs` in Testground, so that it is clear who started a given job.

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -38,6 +38,18 @@ func New(cfg *config.EnvConfig) (srv *Daemon, err error) {
 
 	r := mux.NewRouter()
 
+	// Authentication
+	r.Use(func(next http.Handler) http.Handler {
```
Comment (raulk) [inline pkg/daemon/daemon.go:42]: We should add this handler only if there are tokens in the cfg.

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -38,6 +38,18 @@ func New(cfg *config.EnvConfig) (srv *Daemon, err error) {
...
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (raulk) [inline pkg/daemon/daemon.go:43]: We can initialise a map[string]struct{} inside the closure, and close over it (capture it) in the nested closure. Then we can get rid of the contains method. BTW -- when dealing with critical config values like this, I really recommend cleaning up the user-input strings using strings.TrimSpace.

File: pkg/client/client.go [hunk 3]
```diff
@@ -396,6 +396,9 @@ func (c *Client) request(ctx context.Context, method string, path string, body i
 	req, err := http.NewRequest(method, "http://"+c.endpoint+path, body)
 	req = req.WithContext(ctx)
 
+	// add auth token
+	req.Header.Add("X-Token", c.cfg.Client.Token)
```
Comment (raulk) [inline pkg/client/client.go:400]: BTW, I'd recommend using the standard `Authorization: Bearer` convention: https://swagger.io/docs/specification/authentication/bearer-authentication/.

File: pkg/daemon/daemon.go (hunk 1 above)
Comment (nonsense) [reply pkg/daemon/daemon.go:42]: Right, I added check for len, but inside the handler. You're right that it should be for the handler itself.

**Pull request merged**